### DISCUSSION TITLE
Server Actions as a GET Request & Using stringified argument

### DISCUSSION THREAD

**Discussion by vicnz** [vercel/next.js#58847] [Help] [answered]
### Summary

Can Server Actions be used as a GET Request, I'm sending `JSON.stringifiy()` as the argument passed in the `server action` instead of FormData Object. and returning a `JSON.stringify` for replies.

### Additional information

```js
'use server'
export const action = async (data: string) => {
    let converted = JSON.parse(data)
    //DO Server Side Operation here
    return JSON.stringify(returnValue)
}
```

**Comment by ddddami** [comment, marked as answer]
Yes, Server Actions can be used as a GET Request. In fact, they are the preferred way to handle GET requests in Next.js 13, as they provide a more flexible and performant alternative to traditional API routes.

To use a Server Action for a GET request, you simply need to make a GET request to the endpoint that corresponds to the Server Action. For example, if you have a Server Action named myAction that takes a JSON-formatted string as input, you would make a GET request to the following endpoint:

/api/my-action?data=<your-json-data>
The JSON-formatted data would be sent as a query parameter in the URL. The Server Action would then receive the data parameter, parse it into a JSON object, and perform the desired operation. The Server Action would then return a JSON-formatted response, which would be returned to the client.

Here's an example of how to implement a Server Action that handles a GET request and returns a JSON-formatted response:

```javascript
import type { NextRequest } from 'next/server';

export function middleware(req: NextRequest) {
  if (req.nextUrl.pathname === '/api/my-action') {
    const data = req.query.data;
    const converted = JSON.parse(data);

    // Perform server-side operation here

    const returnValue = {
      // ... return values
    };

    return NextResponse.json(JSON.stringify(returnValue));
  }
}
```
This middleware function would intercept GET requests to the /api/my-action endpoint, parse the JSON-formatted data from the query parameter, perform the desired operation, and return a JSON-formatted response.

Using Server Actions for GET requests is a more modern and efficient approach to handling API requests in Next.js 13. They provide a cleaner separation of concerns, better performance, and easier maintenance.
